near-lake-framework/src/lib.rs

```diff
@@ -0,0 +1,270 @@
+//!
+//!`~/.aws/credentials`
+//!```
+//![default]
+//!aws_access_key_id=AKIAIOSFODNN7EXAMPLE
+//!aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+//!```
+//!
+//![AWS docs: Configuration and credential file settings](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
+//!
+//!### Dependencies
+//!
+//!Add the following dependencies to your `Cargo.toml`
+//!
+//!```toml
+//!...
+//![dependencies]
+//!futures = "0.3.5"
+//!itertools = "0.10.3"
+//!tokio = { version = "1.1", features = ["sync", "time", "macros", "rt-multi-thread"] }
+//!tokio-stream = { version = "0.1" }
+//!
+//!# NEAR Lake Framework
+//!near-lake-framework = "0.3.0"
+//!```
+//!
+//! ### Custom S3 storage
+//!
+//! In case you want to run your own [near-lake](https://github.com/near/near-lake) instance and store data in some S3 compatible storage ([Minio](https://min.io/) or [Localstack](https://localstack.cloud/) as example)
+//! You can owerride default S3 API endpoint by using `s3_endpoint` option
+//!
+//! - run minio
+//!
+//! ```bash
+//! $ mkdir -p /data/near-lake-custom && minio server /data
+//! ```
+//!
+//! - pass custom `aws_sdk_s3::config::Config` to the [LakeConfigBuilder]
+//!
+//! ```rust
+//! use aws_sdk_s3::Endpoint;
+//! use http::Uri;
+//! use near_lake_framework::LakeConfigBuilder;
+//!
+//! # async fn main() {
+//! let aws_config = aws_config::from_env().load().await;
+//! let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config);
+//! s3_conf = s3_conf
+//!     .endpoint_resolver(
+//!             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
+//!     .build();
+//!
+//! let config = LakeConfigBuilder::default()
+//!     .s3_config(s3_conf)
+//!     .s3_bucket_name("near-lake-data-custom")
+//!     .start_block_height(1)
+//!     .build()
+//!     .expect("Failed to build LakeConfig");
+//! # }
+//! ```
+//!
+//! ## Configuration
+//!
+//! Everything should be configured before the start of your indexer application via `LakeConfigBuilder` struct.
+//!
+//! Available parameters:
+//!
+//! * [`start_block_height(value: u64)`](LakeConfigBuilder::start_block_height) - block height to start the stream from
+//! * *optional* [`s3_bucket_name(value: impl Into<String>)`](LakeConfigBuilder::s3_bucket_name) - provide the AWS S3 bucket name (you need to provide it if you use custom S3-compatible service, otherwise you can use [LakeConfigBuilder::mainnet] and [LakeConfigBuilder::testnet])
+//! * *optional* [`LakeConfigBuilder::s3_region_name(value: impl Into<String>)`](LakeConfigBuilder::s3_region_name) - provide the AWS S3 region name (if you need to set a custom one)
+//! * *optional* [`LakeConfigBuilder::s3_config(value: aws_sdk_s3::config::Config`](LakeConfigBuilder::s3_config) - provide custom AWS SDK S3 Config
+//!
+//! ## Cost estimates
+//!
+//! **TL;DR** approximately $18.15 per month (for AWS S3 access, paid directly to AWS) for the reading of fresh blocks
```

Comment:
As far as I know, we have the plan to close everything with our keys and charge extra for the access.
@pkudinov could you please clarify?

Reply:
Disregard the docs, for now, they are not updated.

I do remember the talks to serve everything with the Pagoda keys. It is not implemented on the infra side yet, meanwhile, we continue to support direct access to AWS S3 through the developers' keys. And I believe we won't disable it anyway.